presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/type/AbstractPrestoTimestampEncoder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.base.type;
+
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.type.TimestampType;
+import org.joda.time.DateTimeZone;
+
+abstract class AbstractPrestoTimestampEncoder<T extends Comparable<T>>
```

Comment:
Comparable does not make any sense from `PrestoTimestampEncoder` perspective (just lifts eyebrows up), but is needed here for purely technical reasons. I have commented in the places where you leverage this so that you can remove this from here & simplify the code.